### PR TITLE
[PD-2036]  docker scripts

### PR DESCRIPTION
**Pull request by darrint**
These are the docker scripts I wrote for myself. This system has proven useful even with it's flaws.

Future improvements might include:

* Remove darrint-isms from dockerdev.sh. (tag names, etc)
* Make background and backgroundstop commands more "symmetric".
* Separate build and start of background containers.
* Deploy some kind of local infrastructure in office so we don't have to repeatedly download and build the same images on everyone's workstation. (That would bring new system setup time down to under 20 minutes, easy)

### CODE REVIEW

File: doc/source/developers/index.rst [hunk 1]
```diff
@@ -9,6 +9,7 @@ Developer Documentation
   mypartners/index
   postajob/index
   seo/index
+  docker/index
```
Comment (aspidites) [inline doc/source/developers/index.rst]: I'd rather this went somewhere else, as at the moment it just looks shoved in the middle of chapters that map cleanly to apps.

File: doc/source/developers/docker/use.rst [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+* There will be permissions issues. The user running tools in the docker
+  container will not have the same uid as the developer's own uid. This will
+  result in occasional problems. This tends to affect pyc files, (not a big
+  deal) and node_modules (occasionally annoying).
+
+What you Need
+=============
+
+* MyJobs should be checked out.
+* A development database backup to restore.
```
Comment (aspidites) [inline doc/source/developers/docker/use.rst]: Restore? Isn't this a new set up? Where would said backup come from?

File: doc/source/developers/docker/use.rst [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+* An archive of a working Solr configuration.
```
Comment (aspidites) [inline doc/source/developers/docker/use.rst]: Again, where do these archives come from?  We have a solr setup doc, but that's not an archive, so I'm not sure what's meant here.

File: doc/source/developers/docker/use.rst [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+* Django secrets files, settings files, and a dev_settings file.
```
Comment (aspidites) [inline doc/source/developers/docker/use.rst]: If you followed @fezick's setup instructions, you won't have settings and dev_settings. Instead you'll have whatever is required for the fast switching.

File: doc/source/developers/docker/use.rst [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+------------------
+
+Install docker from your distro.
+
+Know your local ip address. We'll call it $myip from now on.
+
+Terminal Setup (OSX)
+--------------------
+
+Run these commands in every shell:
```
Comment (aspidites) [inline doc/source/developers/docker/use.rst]: In every shell? Should I have multiple terminals open by this point?

File: doc/source/developers/docker/use.rst [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+``eval "$(docker-machine env default)"``
+
+Terminal Setup (Both)
+---------------------
+
+``cd $myjobs`` where myjobs is wherever you checked out the MyJobs code.
+
+``. dk2/env.sh`` This makes the ``dk`` convenience function available.
```
Comment (aspidites) [inline doc/source/developers/docker/use.rst]: Dot's are hard to read. I'd rather we said `source` here. Anyone who knows there shell will know they can replace it with a dot.

**Comment by aspidites** [comment]
Meh. Deleted my documentation comments. I think that if I ever go through this, it'll be easier to update then. I do still think this is sitting in the wrong place in the TOC though. Leaving this for someone else whose actually used the instructions to merge.

File: doc/source/developers/index.rst (hunk 1 above)
Comment (darrint) [inline doc/source/developers/index.rst]: Where do you want it? I put it after the chapters that map cleanly to apps.

Comment (aspidites) [inline doc/source/developers/index.rst]: So it is. I didn't realize that all the set up docs were still in the private docs repo, or I'd have suggested with those. At some point, I think I'd like a "setup" directory that included all instructions for environment setup.

**Comment by aspidites** [comment]
Is there significance in the `dk2` name? It certainly isn't very descriptive.

**Comment by darrint** [comment]
My first cut was called `dk`. This replaced it.

**Comment by lemniscate613** [comment]
With Darrin's help, all of this is working for me locally (except for microsites).

Here are suggested additions to documentation. @darrint see the ?????? below.

```
Preparation
=============

* Clone MyJobs repo.
* Add these to the root of the repo (they are not version controlled):
  * Dump of development MySQL database (named dbbackup.sql). For example:
    ```
    mysqldump -u root -p --all-databases > dbbackup.sql;
    ```
    Beware, production uses MySql 5.5. If you make a dump from a newer version
    you may have a time.
  * An archive of a working Solr configuration. For example:
    ```
    cp -r /usr/local/Cellar/solr/5.3.1/server/solr ~/projects/MyJobs
    ```
    Beware, production uses SOLR ??????. If you make a dump from a newer version
    you may have a time.
  * Get these from another developer:
    * Settings.py
    * Dev_settings.py
    * Secrets.py
```

**Comment by lemniscate613** [comment]
Once this passes tests on Jenkins, I'll merge it. It's still a work in progress. Three of us are using it, but when the 4th person follows the instructions they will probably need to tweak these instructions: 

doc/source/developers/docker/use.rst

**Pull request merged**